test/server/cards/02_SHD/upgrades/TheMandaloriansRifle.spec.ts

```diff
@@ -0,0 +1,94 @@
+describe('The Mandalorian\'s Rifle', function () {
+    integration(function (contextRef) {
+        describe('The Mandalorian\'s Rifle\'s ability', function () {
+            it('should not capture anyone because attached unit is not The Mandalorian', function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        hand: ['the-mandalorians-rifle'],
+                        groundArena: ['the-mandalorian#wherever-i-go-he-goes', 'battlefield-marine'],
+                        spaceArena: ['green-squadron-awing']
+                    },
+                    player2: {
+                        groundArena: ['wampa', { card: 'atst', exhausted: true }, { card: 'scout-bike-pursuer', exhausted: true }]
+                    }
+                });
+
+                const { context } = contextRef;
+
+                context.player1.clickCard(context.theMandaloriansRifle);
+
+                // able to choose non-vehicle friendly unit
+                expect(context.player1).toBeAbleToSelectExactly([context.battlefieldMarine, context.theMandalorian]);
+                context.player1.clickCard(context.battlefieldMarine);
+
+                // not the mandalorian, nothing happen
+                expect(context.wampa).toBeInZone('groundArena');
+                expect(context.atst).toBeInZone('groundArena');
+                expect(context.scoutBikePursuer).toBeInZone('groundArena');
+
+                expect(context.battlefieldMarine).toBeInZone('groundArena');
+                expect(context.theMandalorian).toBeInZone('groundArena');
+                expect(context.player2).toBeActivePlayer();
+            });
+
+            it('should capture an exhausted enemy unit because attached unit is The Mandalorian (unit)', function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        hand: ['the-mandalorians-rifle'],
+                        groundArena: ['the-mandalorian#wherever-i-go-he-goes', 'battlefield-marine'],
+                    },
+                    player2: {
+                        groundArena: ['wampa', { card: 'atst', exhausted: true }, { card: 'scout-bike-pursuer', exhausted: true }]
```

Comment:
Let's add a deployed enemy leader just to confirm the targeting works since capture is new